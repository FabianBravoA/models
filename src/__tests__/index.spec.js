const { validate } = require('../index');


describe('validate', () => {
  it('should invoke callback with mongoose errors', (done) => {
    validate('Project', {
      slug: 'cipher',
      repo: 'Laboratoria/curricula-js',
      path: 'projects/01-cipher',
    }, (err) => {
      expect(err.errors).toMatchSnapshot();
      done();
    });
  });

  it('should return promise when no callback is passed', () => (
    validate('Project', {
      slug: 'cipher',
      repo: 'Laboratoria/curricula-js',
      path: 'projects/01-cipher',
    }).catch((err) => {
      expect(err.errors).toMatchSnapshot();
    })
  ));
});
