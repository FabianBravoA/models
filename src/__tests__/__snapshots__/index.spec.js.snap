// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate should invoke callback with mongoose errors 1`] = `
Object {
  "locale": [ValidatorError: Path \`locale\` is required.],
  "parserVersion": [ValidatorError: Path \`parserVersion\` is required.],
  "rubric": [ValidatorError: Path \`rubric\` is required.],
  "title": [ValidatorError: Path \`title\` is required.],
  "track": [ValidatorError: Path \`track\` is required.],
  "version": [ValidatorError: Path \`version\` is required.],
}
`;

exports[`validate should return promise when no callback is passed 1`] = `
Object {
  "locale": [ValidatorError: Path \`locale\` is required.],
  "parserVersion": [ValidatorError: Path \`parserVersion\` is required.],
  "rubric": [ValidatorError: Path \`rubric\` is required.],
  "title": [ValidatorError: Path \`title\` is required.],
  "track": [ValidatorError: Path \`track\` is required.],
  "version": [ValidatorError: Path \`version\` is required.],
}
`;
